modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/context/provider/PaginationContextProvider.java

```diff
@@ -22,21 +28,69 @@
 @Provider
 public class PaginationContextProvider implements ContextProvider<Pagination> {
 
+	public PaginationContextProvider(
+		ConfigurationProvider configurationProvider, Portal portal) {
+
+		_configurationProvider = configurationProvider;
+		_portal = portal;
+	}
+
 	@Override
 	public Pagination createContext(Message message) {
 		HttpServletRequest httpServletRequest =
 			ContextProviderUtil.getHttpServletRequest(message);
 
 		String page = httpServletRequest.getParameter("page");
+
+		if ((page != null) && (GetterUtil.getLong(page) <= 0)) {
+			throw new InvalidPaginationException(
+				String.format(
+					"The Page parameter introduced [%s] is not valid. Only " +
+						"numbers higher or equal to 1 are accepted.",
+					page));
+		}
+
 		String pageSize = httpServletRequest.getParameter("pageSize");
+		long pageSizeValueIfNotSpecified = 20L;
+
+		try {
+			HeadlessAPICompanyConfiguration headlessAPICompanyConfiguration =
+				_configurationProvider.getCompanyConfiguration(
+					HeadlessAPICompanyConfiguration.class,
+					_portal.getCompanyId(httpServletRequest));
 
-		if (StringUtil.equals(page, "0") || StringUtil.equals(pageSize, "0")) {
-			return null;
+			Long maxPageSize =
+				headlessAPICompanyConfiguration.getPaginationSizeLimit();
+
+			if ((maxPageSize < 20) && (maxPageSize > 0)) {
+				pageSizeValueIfNotSpecified = maxPageSize;
+			}
+
+			if ((pageSize != null) && (maxPageSize > 0)) {
+				long pageSizeLong = GetterUtil.getLong(pageSize);
+
+				if ((pageSizeLong > maxPageSize) || (pageSizeLong <= 0)) {
+					throw new InvalidPaginationException(
+						String.format(
+							"The PageSize introduced [%s] exceeds the " +
+								"maximum PageSize configured [%s].",
+							pageSize, maxPageSize));
+				}
+			}
+		}
+		catch (ConfigurationException configurationException) {
+			_log.error(configurationException);
```

Comment:
I am not sure if we should ignore this exception

Reply:
I'm not sure either. The main reason I opted to do this is becouse I saw many examples of a `catch` for Configuration exceptions that only did a `log,error`